rules/0133.go

```diff
@@ -0,0 +1,227 @@
+					problems = append(problems, lint.Problem{
+						Message:    "The `parent` field on Create RPCs should be a string",
+						Descriptor: parentField,
+					})
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// Create methods should not have unrecognized fields.
+func checkCreateRequestMessageUnknownFields() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0133", "request_message", "unknown_fields"),
+			Description:  "check that there are no unknown fields",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/133#request-message",
+		},
+		Callback: descriptor.Callbacks{
+			MessageCallback: func(m p.MessageDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about Get methods for the purpose of this rule;
+				// ignore everything else.
+				if !isCreateRequestMessage(m) {
+					return
+				}
+
+				// Determine the name of the resource being created.
+				resource := regexp.MustCompile("^Create([A-Za-z0-9]+)Request$").FindStringSubmatch(string(m.Name()))[1]
+				resourceSnake := strcase.SnakeCase(resource)
+
+				// Rule check: Establish that there are no unexpected fields.
+				allowedFields := map[string]struct{}{
+					"parent":                            {}, // AIP-133
+					resourceSnake:                       {}, // AIP-133
+					fmt.Sprintf("%s_id", resourceSnake): {}, // AIP-133
+					"request_id":                        {}, // AIP-155
+				}
+				fields := m.Fields()
+				for i := 0; i < fields.Len(); i++ {
+					field := fields.Get(i)
+					if _, ok := allowedFields[string(field.Name())]; !ok {
+						problems = append(problems, lint.Problem{
+							Message: fmt.Sprintf(
+								"Create RPCs should only only contain fields explicitly described in AIPs, not %q.",
+								string(field.Name()),
+							),
+							Descriptor: field,
+						})
+					}
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// Get messages should use the resource (or Operation) as the response message
+func checkCreateResponseMessageName() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0133", "response_message", "name"),
+			Description:  "check that Create RPCs have appropriate response messages",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/133#guidance",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about Get methods for the purpose of this rule;
+				// ignore everything else.
+				if !isCreateMethod(m) {
+					return
+				}
+
+				// Sanity check: If the response type is an LRO, accept this.
+				// TODO: Check the response type annotation.
+				responseMessageName := string(m.Output().Name())
+				if responseMessageName == "Operation" {
```

Comment:
Is the package stripped for responseMessageName? Given:

`rpc CreateFoo(CreateFooRequest) returns (google.longrunning.Operation)`

Reply:
Yes, it is, although I since learned that `FullName` exists and would do what I want. I will update this.

Reply:
Done.